lib/src/entrypoint.dart

```diff
@@ -373,6 +379,15 @@ class Entrypoint {
             '${p.basename(executable.relativePath)}-$versionSuffix.snapshot');
   }
 
+  String incrementalDillPathOfExecutable(Executable executable) {
+    assert(p.isRelative(executable.relativePath));
+    return isGlobal
+        ? p.join(_incrementalDillsPath,
+            '${p.basename(executable.relativePath)}.incremental.dill')
+        : p.join(_incrementalDillsPath, executable.package,
+            '${p.basename(executable.relativePath)}.incremental.dill');
```

Comment:
Are we sure we don't want to inject SDK version number into the filename... like `.dart-a.b.c.`?

How will behave with SDK upgrades, or downgrades?

Reply:
When upgrading/downgrading the sdk it still does a full recompile internally (I believe this is detected by the kernel version of the incremental dill) - cc @jonahwilliams can maybe confirm?

Reply:
But will it throw an error? or just overwrite the old invalid `.dill` file?

Reply:
We will just overwrite the old file with a new one, it handles it gracefully.

Reply:
It would be ideal to have some sort of test around this to ensure it continues to work as expected.

I am not sure what the easiest way would be to actually write that test (how would we detect "failure" consistently?).

We could keep files around with sdk version numbers in them but we would want some way of cleaning up old ones in that case. Maybe just remember the previous sdk version (store it in a file?) and then bash away the entire `incremental` dir if it changes?

Reply:
I think such tests would be fun to do... but we don't have that at the moment, so let's do that in a separate PR another day.